src/core/server/saved_objects/saved_objects_type_registry.ts

```diff
@@ -33,7 +33,7 @@ export class SavedObjectTypeRegistry {
       throw new Error(`Type '${type.name}' is already registered`);
     }
     validateType(type);
-    this.types.set(type.name, deepFreeze(type));
+    this.types.set(type.name, deepFreeze(type) as SavedObjectsType);
```

Comment:
What change caused this cast to be required?

Reply:
I believe it's the fact that `SavedObjectsFieldMapping` is the union type from now on.